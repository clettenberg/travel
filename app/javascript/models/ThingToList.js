class ThingToList {
  constructor (url, title, id) {
    this.url = url
    this.title = title
    this.id = id
  }
}

export default ThingToList
