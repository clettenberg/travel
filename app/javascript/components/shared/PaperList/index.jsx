import React from 'react'
import PropTypes from 'prop-types'

import {
  List,
  ListItem,
  ListItemText,
  ListSubheader,
  ListItemSecondaryAction,
  Divider,
  makeStyles,
  IconButton
} from '@material-ui/core'
import AddIcon from '@material-ui/icons/Add'
import ThingToList from '../../../models/ThingToList'

const propTypes = {
  title: PropTypes.string.isRequired,
  actionButtonUrl: PropTypes.string.isRequired,
  thingsToList: PropTypes.arrayOf(PropTypes.instanceOf(ThingToList))
}

const ListItemLink = (props) => (
  <ListItem button component='a' {...props} />
)

const useStyles = makeStyles({
  root: {
    backgroundColor: 'white'
  },
  subheader: {
    fontSize: '1.3rem',
    padding: 0,
    '& #heading': {
      marginLeft: '10px'
    }
  }
})

const PaperList = ({ title, actionButtonUrl, actionButtonText, thingsToList }) => {
  const classes = useStyles()
  return (
    <List
      className={classes.root}
      subheader={
        <ListSubheader component='div' className={classes.subheader}>
          <div id='heading'>{title}</div>
          <ListItemSecondaryAction>
            <IconButton
              color='primary'
              href={actionButtonUrl}
              aria-label={actionButtonText}
            >
              <AddIcon />
            </IconButton>
          </ListItemSecondaryAction>
          <Divider />
        </ListSubheader>
      }
    >
      {thingsToList.map(({ url, title, id, date }) => (
        <ListItemLink href={url} key={id}>
          <ListItemText primary={title} secondary={date} />
        </ListItemLink>
      ))}
    </List>
  )
}

PaperList.propTypes = propTypes
export default PaperList
