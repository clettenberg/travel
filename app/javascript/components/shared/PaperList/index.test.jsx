import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import PaperList from '.'
import ThingToList from '../../../models/ThingToList'

describe('PaperList', () => {
  describe('when provided with things to list', () => {
    it('should render', () => {
      const thingsToList = [
        { title: 'title 1', url: '/url/1', id: 1 },
        { title: 'title 2', url: '/url/2', id: 2 },
        { title: 'title 3', url: '/url/3', id: 3 }
      ].map(({ title, url, id }) => new ThingToList(url, title, id))

      const wrapper = shallow(
        <PaperList
          title='Testing'
          actionButtonUrl='/test/it'
          thingsToList={thingsToList}
        />
      )
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })
})
