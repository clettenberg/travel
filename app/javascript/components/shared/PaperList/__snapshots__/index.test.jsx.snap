// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaperList when provided with things to list should render 1`] = `
<WithStyles(ForwardRef(List))
  className="makeStyles-root-1"
  subheader={
    <WithStyles(ForwardRef(ListSubheader))
      className="makeStyles-subheader-2"
      component="div"
    >
      <div
        id="heading"
      >
        Testing
      </div>
      <WithStyles(ForwardRef(ListItemSecondaryAction))>
        <WithStyles(ForwardRef(IconButton))
          aria-label="Test It"
          color="primary"
          href="/test/it"
        >
          <Memo(AddIcon) />
        </WithStyles(ForwardRef(IconButton))>
      </WithStyles(ForwardRef(ListItemSecondaryAction))>
      <WithStyles(ForwardRef(Divider)) />
    </WithStyles(ForwardRef(ListSubheader))>
  }
>
  <ListItemLink
    href="/url/1"
    key="1"
  >
    <WithStyles(ForwardRef(ListItemText))
      primary="title 1"
    />
  </ListItemLink>
  <ListItemLink
    href="/url/2"
    key="2"
  >
    <WithStyles(ForwardRef(ListItemText))
      primary="title 2"
    />
  </ListItemLink>
  <ListItemLink
    href="/url/3"
    key="3"
  >
    <WithStyles(ForwardRef(ListItemText))
      primary="title 3"
    />
  </ListItemLink>
</WithStyles(ForwardRef(List))>
`;
